airbyte_cdk/sources/declarative/partition_routers/substream_partition_router.py

```diff
@@ -131,6 +143,42 @@ def _get_request_option(
                         parent_config.request_option.inject_into_request(params, value, self.config)
         return params
 
+    def process_parent_record(
```

Comment:
I'm wondering if we could have kept this internal to SubstreamPartitionRouter if we would have had the `lazy_read_pointer` on the SubstreamPartitionRouter level and it could have set extra_fields as:
```
{
  <partition_field>: <partition_value>,
  "parent_slice": <parent_partition>,
  "extra_fields": { "child_response": <response> }
}
```

The benefits would be that:
* we can leverage concurrency which would be useful in terms of performance if many slices require pagination
* the logic of looping over the stream slices would still be hidden as part of the SubstreamPartitionRouter and therefore probably easier to maintain

We would still need to have a new retriever though but it would analyze the first page as being `slice.extra_fields.child_response`.

Reply:
Nice proposal!

I’ve moved child response handling to `partition_router`, update `LazySimpleRetriever` and move `lazy_read_pointer` to `parent_config`. Now, if a user has multiple sources/parents for child records, they can set up lazy_read_pointer for each one individually.